File: app/src/main/java/com/fittrack/app/data/entity/WorkoutSet.kt
package com.fittrack.app.data.entity

import androidx.room.Entity
import androidx.room.ForeignKey
import androidx.room.Index
import androidx.room.PrimaryKey

@Entity(
    tableName = "workout_sets",
    foreignKeys = [
        ForeignKey(
            entity = WorkoutExercise::class,
            parentColumns = ["id"],
            childColumns = ["workoutExerciseId"],
            onDelete = ForeignKey.CASCADE
        )
    ],
    indices = [Index("workoutExerciseId")]
)
data class WorkoutSet(
    @PrimaryKey(autoGenerate = true) val id: Long = 0,
    val workoutExerciseId: Long,
    val setNumber: Int,
    val reps: Int,
    val weightKg: Double,
    val isWarmup: Boolean = false,
    val rpe: Int? = null
)

File: ios/FitTrack/FitTrack/Models/WorkoutSet.swift
import Foundation
import SwiftData

@Model
final class WorkoutSet {
    var setNumber: Int
    var reps: Int
    var weightKg: Double
    var isWarmup: Bool
    var rpe: Int?
    var workoutExercise: WorkoutExercise?

    init(setNumber: Int, reps: Int, weightKg: Double, isWarmup: Bool = false, rpe: Int? = nil) {
        self.setNumber = setNumber
        self.reps = reps
        self.weightKg = weightKg
        self.isWarmup = isWarmup
        self.rpe = rpe
    }
}
